tools/ci_build/github/apple/build_ios_framework.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python3
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import argparse
+import os
+import pathlib
+import json
+import subprocess
+import sys
+import glob
+import shutil
+
+SCRIPT_DIR = os.path.dirname(os.path.realpath(__file__))
+REPO_DIR = os.path.normpath(os.path.join(SCRIPT_DIR, "..", "..", "..", ".."))
+BUILD_PY = os.path.join(REPO_DIR, "tools", "ci_build", "build.py")
+
+# We by default will build below 2 archs
+DEFAULT_BUILD_OSX_ARCHS = [["iphonesimulator", "x86_64"], ["iphoneos", "arm64"]]
+
+
+def _parse_build_settings(args):
+    _setting_file = args.build_settings_file.resolve()
+
+    if not _setting_file.is_file():
+        raise FileNotFoundError('Build config file {} is not a file.'.format(_setting_file))
+
+    with open(_setting_file) as f:
+        _build_settings_data = json.load(f)
+
+    build_settings = {}
+
+    if 'build_osx_archs' in _build_settings_data:
+        build_settings['build_osx_archs'] = []
+        for _arch in _build_settings_data['build_osx_archs']:
+            build_settings['build_osx_archs'].append([_arch["sysroot"], _arch["arch"]])
+    else:
+        build_settings['build_osx_archs'] = DEFAULT_BUILD_OSX_ARCHS
+
+    build_params = []
+    if 'build_params' in _build_settings_data:
+        build_params += _build_settings_data['build_params']
+    else:
+        raise ValueError('build_params is required in the build config file')
+
+    build_settings['build_params'] = build_params
+    return build_settings
+
+
+def _build_package(args):
+    build_settings = _parse_build_settings(args)
+    build_dir = os.path.abspath(args.build_dir)
+
+    # Temp dirs to hold building results
+    _intermediates_dir = os.path.join(build_dir, 'intermediates')
+    _build_config = args.config
+    _base_build_command = [
+        sys.executable, BUILD_PY,
+        '--config=' + _build_config,
+        '--apple_deploy_target=' + args.apple_deploy_target
+    ] + build_settings['build_params']
+
+    # paths of the onnxruntime libraries for different archs
+    _ort_libs = []
+    _info_plist = ''
+
+    # Build binary for each arch, one by one
+    for _osx_arch in build_settings['build_osx_archs']:
+        _build_dir = os.path.join(_intermediates_dir, _osx_arch[0] + "_" + _osx_arch[1])
```

Comment:
>_build_dir [](http://example.com/codeflow?start=8&length=10)

build_dir and _build_dir are similar, maybe choose a more specific name for this?